### ISSUE TITLE
Some GraphQL Clients treat fields as deprecated

### ISSUE THREAD

**Issue by jasonbahl** [wp-graphql/wp-graphql#1632]
For fields that are not deprecated, WPGraphQL currently outputs `deprecationReason` as an empty string. 

Some clients such as GraphQL Voyager see that as a flag that the fields are deprecated. 

WPGraphQL Should instead output a null for fields that are not deprecated and output a string only if there is a valid deprecation reason.

**Comment by dsbrianwebster** [comment]
@jasonbahl I can confirm this is happening on a project we're moving over to WPGraphQL. We're using graphql-codegen to generate typed clients, in this case for `graphql-request` and a large number of fields are being interpreted to be deprecated...

example

```js
/** The post type */
export type Post = Node & ContentNode & DatabaseIdentifier & NodeWithTemplate & UniformResourceIdentifiable & NodeWithTitle & NodeWithContentEditor & NodeWithAuthor & NodeWithFeaturedImage & NodeWithExcerpt & NodeWithRevisions & MenuItemLinkable & {
  __typename?: 'Post';
  /** @deprecated  */
  acfPost?: Maybe<Post_Acfpost>;
  /** @deprecated  */
  acfPostOptions?: Maybe<Post_Acfpostoptions>;
  /**
   * Connection between the NodeWithAuthor type and the User type
   * @deprecated 
   */
  author?: Maybe<NodeWithAuthorToUserConnectionEdge>;
  /**
   * The database identifier of the author of the node
   * @deprecated 
   */
  authorDatabaseId?: Maybe<Scalars['Int']>;
  /**
   * The globally unique identifier of the author of the node
   * @deprecated 
   */
  authorId?: Maybe<Scalars['ID']>;
  /**
   * Connection between the post type and the category type
   * @deprecated 
   */
  categories?: Maybe<PostToCategoryConnection>;
  /**
   * The content of the post.
   * @deprecated 
   */
  content?: Maybe<Scalars['String']>;
  /**
   * Connection between the ContentNode type and the ContentType type
   * @deprecated 
   */
  contentType?: Maybe<ContentNodeToContentTypeConnectionEdge>;
  /**
   * The ID of the node in the database.
   * @deprecated 
   */
  databaseId: Scalars['Int'];
  /**
   * Post publishing date.
   * @deprecated 
   */
  date?: Maybe<Scalars['String']>;
  /**
   * The publishing date set in GMT.
   * @deprecated 
   */
  dateGmt?: Maybe<Scalars['String']>;
  /**
   * The desired slug of the post
   * @deprecated 
   */
  desiredSlug?: Maybe<Scalars['String']>;
  /**
   * If a user has edited the node within the past 15 seconds, this will return the user that last edited. Null if the edit lock doesn&#039;t exist or is greater than 15 seconds
   * @deprecated 
   */
  editingLockedBy?: Maybe<ContentNodeToEditLockConnectionEdge>;
  /**
   * The RSS enclosure for the object
   * @deprecated 
   */
  enclosure?: Maybe<Scalars['String']>;
  /**
   * Connection between the ContentNode type and the EnqueuedScript type
   * @deprecated 
   */
  enqueuedScripts?: Maybe<ContentNodeToEnqueuedScriptConnection>;
  /**
   * Connection between the ContentNode type and the EnqueuedStylesheet type
   * @deprecated 
   */
  enqueuedStylesheets?: Maybe<ContentNodeToEnqueuedStylesheetConnection>;
  /**
   * The excerpt of the post.
   * @deprecated 
   */
  excerpt?: Maybe<Scalars['String']>;
  /**
   * Connection between the NodeWithFeaturedImage type and the MediaItem type
   * @deprecated 
   */
  featuredImage?: Maybe<NodeWithFeaturedImageToMediaItemConnectionEdge>;
  /**
   * The database identifier for the featured image node assigned to the content node
   * @deprecated 
   */
  featuredImageDatabaseId?: Maybe<Scalars['Int']>;
  /**
   * Globally unique ID of the featured image assigned to the node
   * @deprecated 
   */
  featuredImageId?: Maybe<Scalars['ID']>;
  /**
   * The global unique identifier for this post. This currently matches the value stored in WP_Post-&gt;guid and the guid column in the &quot;post_objects&quot; database table.
   * @deprecated 
   */
  guid?: Maybe<Scalars['String']>;
  /**
   * The globally unique identifier of the post object.
   * @deprecated 
   */
  id: Scalars['ID'];
  /**
   * Whether the object is a node in the preview state
   * @deprecated 
   */
  isPreview?: Maybe<Scalars['Boolean']>;
  /**
   * Whether the object is restricted from the current viewer
   * @deprecated 
   */
  isRestricted?: Maybe<Scalars['Boolean']>;
  /**
   * True if the node is a revision of another node
   * @deprecated 
   */
  isRevision?: Maybe<Scalars['Boolean']>;
  /**
   * Whether this page is sticky
   * @deprecated 
   */
  isSticky: Scalars['Boolean'];
  /**
   * The user that most recently edited the node
   * @deprecated 
   */
  lastEditedBy?: Maybe<ContentNodeToEditLastConnectionEdge>;
  /**
   * The permalink of the post
   * @deprecated 
   */
  link?: Maybe<Scalars['String']>;
  /**
   * The local modified time for a post. If a post was recently updated the modified field will change to match the corresponding time.
   * @deprecated 
   */
  modified?: Maybe<Scalars['String']>;
  /**
   * The GMT modified time for a post. If a post was recently updated the modified field will change to match the corresponding time in GMT.
   * @deprecated 
   */
  modifiedGmt?: Maybe<Scalars['String']>;
  /**
   * Connection between the post type and the postFormat type
   * @deprecated 
   */
  postFormats?: Maybe<PostToPostFormatConnection>;
  /**
   * The id field matches the WP_Post-&gt;ID field.
   * @deprecated Deprecated in favor of the databaseId field
   */
  postId: Scalars['Int'];
  /**
   * Connection between the post type and the post type
   * @deprecated 
   */
  preview?: Maybe<PostToPreviewConnectionEdge>;
  /**
   * The database id of the preview node
   * @deprecated 
   */
  previewRevisionDatabaseId?: Maybe<Scalars['Int']>;
  /**
   * Whether the object is a node in the preview state
   * @deprecated 
   */
  previewRevisionId?: Maybe<Scalars['ID']>;
  /**
   * If the current node is a revision, this field exposes the node this is a revision of. Returns null if the node is not a revision of another node.
   * @deprecated 
   */
  revisionOf?: Maybe<NodeWithRevisionsToContentNodeConnectionEdge>;
  /**
   * Connection between the post type and the post type
   * @deprecated 
   */
  revisions?: Maybe<PostToRevisionConnection>;
  /**
   * The Yoast SEO data of the post
   * @deprecated 
   */
  seo?: Maybe<PostTypeSeo>;
  /**
   * The uri slug for the post. This is equivalent to the WP_Post-&gt;post_name field and the post_name column in the database for the &quot;post_objects&quot; table.
   * @deprecated 
   */
  slug?: Maybe<Scalars['String']>;
  /**
   * The current status of the object
   * @deprecated 
   */
  status?: Maybe<Scalars['String']>;
  /**
   * Connection between the post type and the tag type
   * @deprecated 
   */
  tags?: Maybe<PostToTagConnection>;
  /**
   * The template assigned to a node of content
   * @deprecated 
   */
  template?: Maybe<ContentTemplate>;
  /**
   * Connection between the post type and the TermNode type
   * @deprecated 
   */
  terms?: Maybe<PostToTermNodeConnection>;
  /**
   * The title of the post. This is currently just the raw title. An amendment to support rendered title needs to be made.
   * @deprecated 
   */
  title?: Maybe<Scalars['String']>;
  /**
   * URI path for the resource
   * @deprecated 
   */
  uri: Scalars['String'];
};
```

**Comment by jasonbahl** [comment]
@dsbrianwebster can you confirm that this PR (https://github.com/wp-graphql/wp-graphql/pull/1633) fixes it for you?

**Comment by dsbrianwebster** [comment]
@jasonbahl working beautifully with v1.0.5 🥇

**Comment by jasonbahl** [comment]
closed by #1633
